### DISCUSSION TITLE
Why are UNet up blocks channels "flipped"?

### DISCUSSION THREAD

**Discussion by AlejandroBaron** [huggingface/diffusers#7671] [Q&A]
Hello I'm checking the UNet implementation and when instantiating the up blocks, I see the input channels are "flipped" from my understanding of UNets, in the "up" side/second half

https://github.com/huggingface/diffusers/blob/08bf7545070764d41637e90290ca9b92b392263e/src/diffusers/models/unets/unet_2d.py#L209-L214

So, if `block_out_channels` is let's say `(32,64,128)`, then pairs of `input_channel` and `output_channel` would be

|  `input_channel`| `output_channel`|`prev_channels`|
|-----|-----|-----|
| 64  | 128 | 128
| 32  |  64 | 128
| 32  |  32 | 64

Shouldn't the up half of the `UNet` be bottlenecking the channels (i.e. `input_channels>output_channels`? I get that then when instantiating each layer the input channels are  increased by the skip connection/`prev_channels` but still this looks weird to me

EDIT:

To support my theory, I think the way I comment this is how it's done in the StabilityAI repo
https://github.com/Stability-AI/stablediffusion/blob/main/ldm/modules/diffusionmodules/openaimodel.py

**Comment by tolgacangoz** [comment]
I think you understood correctly. What is your question exactly, could you elaborate?

**Reply by AlejandroBaron** [reply]
@standardAI thanks for answering. My question is that the channels should be 

input=>output
128 => 64
64 => 32
32 => 32

but as you can see in my table, it's the "flipped case"
